### ISSUE TITLE
Deleting duplicate question causes errors

### ISSUE THREAD

**Issue by drdrew42** [livinglab/webwork-for-wordpress#106]
I attempted to delete a duplicate question (literally the same question posted twice).

Now the remaining copy of the question does not load properly, and does not show up as "in-progress" either. Also, clicking on "show my problem" no longer reveals the WW problem, and the problem set from which this question originated has been removed from the dropdown menu (it should be ZeroProductProperty). I don't believe there have been other questions asked from this problem set by other users.

LINK:
https://openlab.citytech.cuny.edu/ol-webwork/#:problemId=:questionId=691

This link now shows two questions (from different problem sets - so definitely not from the same original WW problem). The question of interest here is the one from user: proy.

**Comment by drdrew42** [comment]
Hmm, this link no longer seems to show the question I'm talking about - though manually finding the question from the ol-webwork site still has the same link... 

I also noticed this question no longer has the problemsetname | course | section information displayed underneath the question...

![screen shot 2017-09-11 at 9 02 06 pm](https://user-images.githubusercontent.com/3912882/30303239-890b6b28-9734-11e7-94ad-fc516312fd1f.png)

**Comment by boonebgorges** [comment]
Yeah, it seems as if the problem metadata was lost. You can see this in the URL: the problemId parameter is empty. This is causing the other-questions-for-this-problem query to run incorrectly, which is why you're seeing the problems from the wrong sets.

I'll do my best to investigate this fully in the next day or two.

**Comment by boonebgorges** [comment]
Sorry for the delay on this. It does indeed look like the `problem_id` metadata is missing from the database. This means that OL-WW is unable to show the metadata under the problem, and it also explains the malformed URL (which then returns unexpected results).

So, the question is why this metadata is missing. There are a couple places in the POST chain where this could happen. The most likely candidate is here: https://github.com/livinglab/webwork-for-wordpress/blob/master/classes/Server/Util/ProblemFormatter.php#L305 We try to fish the LibraryID out of the POSTed text with a regular expression. In this case, it's possible that either (a) the LibraryID `...pg` was malformed in the original problem text, or (b) the regular expression is too restrictive or has some other sort of error (it only allows alphanumeric chars, hyphens, underscores, and forward slashes).

@drdrew42 A few questions for you:
1. Are you aware of any other questions posted to OL-WW from the same problem? I know this is a hard thing to filter since the problem data isn't be processed properly, but perhaps you can tell by comparing 'View My Problem' text. If there are other questions from the same problem that *don't* exhibit the same issue, then the issue is probably due to a temporary glitch on the WW end rather than a problem with our platform.
2. Is it possible for you to give me access to the source problem on WW? If the source of the issue is what I think it is, it'll be much easier for me to debug with an actual problem.

**Comment by drdrew42** [comment]
Is it possible that the metadata for this question was accidentally removed when I deleted the duplicate copy of this question? Because that is what happened - the metadata for this question was present (for both copies of the same question) prior to deleting the duplicate.

1) No, I'm not aware of other questions from the same WW problem. I know that this question was asked from a new-this-semester problem set, so it did not overlap with questions asked last year. I have tried "ask for help" on each problem from this set, but "proy"s question does not appear on any of them.

1b) I cannot get the "View My Problem" to load for this question. This was not an issue prior to deleting the duplicate.

2) Not exactly, no, but this is not a question from a GeoGebra problem set - if that's what you're suspecting? This question was originally asked (twice) from a problem in the ZeroProductProperty set - I cannot remember which one, and I can no longer see the problem text to determine it now. Though there's nothing out of the ordinary in this problem to distinguish its behavior based on problem content...

e.g. 
https://openlab.citytech.cuny.edu/ol-webwork/#:problemId=local/setZeroProductProperty/ZPP-standard-medium.pg:questionId=728
is a question from this problem set (asked a week later with no issue).

**Comment by drdrew42** [comment]
Timeline for this question is as follows:

1) I notice two copies of the exact same question on ol-ww. (same student text in each box: "my question" and "what I've tried")
2) I respond to one of the two copies.
3) I realize that "delete" is now possible - so I select the un-responded-to copy to delete.
4) After deleting the duplicate, the remaining copy no longer shows my response and has shifted from "in-progress" to "unanswered".
5) I realize that the remaining copy also no longer contains metadata, and neither the question link nor the "show my problem" link work as intended.
6) create issue on github ;P

**Comment by boonebgorges** [comment]
> Is it possible that the metadata for this question was accidentally removed when I deleted the duplicate copy of this question? 

I assume that it does have something to do with it, but I don't really understand how it's happened. If there were actual duplicates - as in, two copies of the same question stored in the database - then deleting one should have nothing to do with the other one. Based on the timeline you've given, one possibility is that there was only one question, but some bug in the display layer was causing it to show up more than once. Still: in this case, I'd expect deleting one of the "duplicates" to delete both altogether, rather than leaving one of them corrupt.

Have you tried creating duplicate questions from WW yourself? Ie: go to a WW problem; click Ask For Help and ask 'foo'; then do the same thing again. Then, in OL-WW, delete one of the duplicates. If so, are you able to reproduce the broken metadata?

**Comment by drdrew42** [comment]
Sure, is our version synced up with the OL-testing site?  Just so that I'm not spamming our production site?

**Comment by boonebgorges** [comment]
The versions of OL-WW should be the same in both places.

**Comment by drdrew42** [comment]
No luck on trying to recreate this error. I did successfully create a duplicate post, but I did not run into any issues (even when using a brand new problem set) when deleting one of the copies...

**Comment by boonebgorges** [comment]
Thanks for testing.

Given that it's not easily reproducible, I'm unsure where to go next. I think we may have to wait for another such instance to crop up in the wild. @drdrew42 If you see more duplicate items on OL-WW, please don't deduplicate - report the details to me right away, and I'll see if I can draw some conclusions.

In the meantime, would it help the current situation if I manually reset the metadata for the broken question? I can do so, if you provide me with the relevant section info and Library ID.

**Comment by drdrew42** [comment]
https://openlab.citytech.cuny.edu/ol-webwork/#:problemId=local/setZeroProductProperty/ZPP-binomials-medium.pg:questionId=743

Please use this question to repopulate the metadata.

**Comment by drdrew42** [comment]
It will have the correct problem set, section, professor, and library ID.

**Comment by boonebgorges** [comment]
@drdrew42 Thank you sir! This should now be fixed. See https://openlab.citytech.cuny.edu/ol-webwork/#:problemId=local/setZeroProductProperty/ZPP-binomials-medium.pg:questionId=691
